ansible_base/lib/dynamic_config/dynamic_settings.py

```diff
@@ -114,3 +114,35 @@
 
 if 'ansible_base.rest_pagination' in INSTALLED_APPS:
     REST_FRAMEWORK['DEFAULT_PAGINATION_CLASS'] = 'ansible_base.rest_pagination.DefaultPaginator'
+
+
+if 'ansible_base.rbac' in INSTALLED_APPS:
+    # Settings for the RBAC system, override as necessary in app
+    ANSIBLE_BASE_ROLE_PRECREATE = {
+        'object_admin': '{cls._meta.model_name}-admin',
+        'org_admin': 'organization-admin',
+        'org_children': 'organization-{cls._meta.model_name}-admin',
+        'special': '{cls._meta.model_name}-{action}',
+    }
+
+    # Permissions a user will get when creating a new item
+    ANSIBLE_BASE_CREATOR_DEFAULTS = ['add', 'change', 'delete', 'view']
+
+    # Specific feature enablement bits
+    # For assignments
+    ANSIBLE_BASE_ALLOW_TEAM_PARENTS = True
+    ANSIBLE_BASE_ALLOW_TEAM_ORG_PERMS = True
+    ANSIBLE_BASE_ALLOW_TEAM_ORG_ADMIN = True
+    # For role definitions
+    ANSIBLE_BASE_ALLOW_CUSTOM_ROLES = True
+    ANSIBLE_BASE_ALLOW_CUSTOM_TEAM_ROLES = False
+    # Allows managing singleton permissions
+    ANSIBLE_BASE_ALLOW_SINGLETON_USER_ROLES = False
+    ANSIBLE_BASE_ALLOW_SINGLETON_TEAM_ROLES = False
+    ANSIBLE_BASE_ALLOW_SINGLETON_ROLES_API = True
+
+    # User flags that can grant permission before consulting roles
+    ANSIBLE_BASE_BYPASS_SUPERUSER_FLAGS = ['is_superuser']
```

Comment:
We don't do a superuser role. AWX did, and it synced superuser role membership with the `is_superuser` "user flag". One of my specific objectives was to kill that off.

Not your question, but the setting right below this existed to support auditor, which is migrated to a role, and that needs to die, so I filed https://github.com/ansible/django-ansible-base/issues/220

I think we should still keep `ANSIBLE_BASE_BYPASS_SUPERUSER_FLAGS` because an app _could_ want to use `is_staff`, and who am I to tell them they're wrong?

To your larger point, jump to a world where we manage "remote" permissions. Yes, you could have a "System Admin" role for the app, which is inferior to `is_superuser`, because it gets synced between multiple services. In case it wasn't clear, `is_superuser` will bypass all other checks. On that :+1: I agree strategically.